crates/core/tedge_api/src/device_id.rs

```diff
@@ -0,0 +1,36 @@
+pub fn new_get_child_id_from_topic(parent_device_name: String, topic: String) -> Option<String> {
```

Comment:
The function name and signature need to change as it is a bit misleading.

Maybe something like:

```
get_external_identity_from_topic(device_common_name: String, topic: String)
```

Reply:
renamed it